### PR TITLE
Search feature

### PR DESCRIPTION
**Pull request by Petrimus**
Search feature for finnish and english languages. Implemented with Gatsby-local-search plugin and flexSearch search engine.

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -1,11 +1,82 @@
-module.exports = {
-  siteMetadata: {
+module.exports = { 
+  pathPrefix: "/fullstack-hy2020.github.io",
```
Comment (Kaltsoon) [inline gatsby-config.js:2]: Is this necessary anymore?

File: gatsby-config.js [hunk 2]
```diff
@@ -1,11 +1,82 @@
...
+  siteMetadata: {   
     title: 'Full Stack open 2020',
     description: '',
     author: 'Houston Inc. Consulting oy',
     siteUrl: 'https://fullstack-hy2020.github.io/',
   },
   plugins: [
+    {
```
Comment (Kaltsoon) [inline gatsby-config.js:10]: Maybe we could get rid of the copy-paste code by defining and using a function such as `createSearchConfig(name, lang)` this file? At first glance `name` and query's `lang` argument seems to require to be passed as arguments

File: package.json [hunk 3]
```diff
@@ -53,6 +57,7 @@
     "start": "npm run develop",
     "format": "prettier --write \"src/**/*.js\"",
     "deploy": "gatsby build && gh-pages -d public -b master",
+    "deploy:private": "gatsby build --prefix-paths && gh-pages -d public",
```
Comment (Kaltsoon) [inline package.json:60]: Is this new script necessary?

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -69,23 +69,30 @@ class Header extends Component {
...
+          )}        
         </div>
       </div>
     );
   }
-}
+};
 
 export default Header;
+/*
```
Comment (Kaltsoon) [inline src/components/Header/Header.js:96]: Some commented code

File: src/components/Search/SearchResults.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      </Element>
+    )
+  }
+}
+
+
+export default SearchResults;
+
+/*
+.search-results-container {
```
Comment (Kaltsoon) [inline src/components/Search/SearchResults.js:63]: Some commented css

File: gatsby-config.js (hunk 1 above)
Comment (Petrimus) [reply gatsby-config.js]: Removed

File: package.json (hunk 3 above)
Comment (Petrimus) [reply package.json]: Removed

File: src/components/Header/Header.js (hunk 4 above)
Comment (Petrimus) [reply src/components/Header/Header.js]: Removed

File: src/components/Search/SearchResults.js (hunk 5 above)
Comment (Petrimus) [reply src/components/Search/SearchResults.js]: Removed

File: gatsby-config.js (hunk 2 above)
Comment (Petrimus) [reply gatsby-config.js]: I added create function and tested build. Also added Footer Element to search pages. It seemed that other pages had it also.

**Pull request merged**